hack/build-go.sh

```diff
@@ -71,13 +71,27 @@ for arg in $args; do
     elif [ "${target}" = "install" ]; then
         eval "$(go env)"
         BIN_NAME=$(basename $arg)
-        ARCHBIN=${BIN_NAME}-$(git describe --always --tags)-${GOHOSTOS}-${GOHOSTARCH}
+        ARCH_BASENAME=${BIN_NAME}-$(git describe --always --tags)
+        ARCHBIN=${ARCH_BASENAME}-${GOHOSTOS}-${GOHOSTARCH}
         mkdir -p ${CMD_OUT_DIR}/${BIN_NAME}
         (
             cd $arg
             go vet ./...
             go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCHBIN}
             (cd ${CMD_OUT_DIR}/${BIN_NAME} && ln -sf ${ARCHBIN} ${BIN_NAME})
+
+            # build virtctl binaries for other operating systems
+            if [ "${BIN_NAME}" = "virtctl" ]; then
```

Comment:
One of these will override the binary built at line 80 IIUC (in case of virtctl). The building could therefore be done as a part of an `else` block, and linked afterwards.

Reply:
If you run this script on another architecture than amd64, or another OS than linux/darwin/windows, the build at line 80 wouldn't be overridden. I didn't want to break the "feature" that you always have a binary at the end, which runs on the same platform where you built it. Not sure if that is more a theoretical feature only though... :)

Reply:
Thanks for your contribution @slintes! That said, I think @mpolednik has a point here--that `go build` command will be run twice under the normal use case, which isn't ideal. Like @mpolednik said, couldn't we use if/else or case statements to accomplish a default?

Reply:
mh, no... for virtctl it even runs up to 4 times: once for the os/arch on which the build runs ($GOHOSTOS / $GOHOSTARCH), which CAN at least theoretically be different than linux/amd64, if you run the script directly and not within docker. And then I added builds for linux/darwin/windows on amd64, because IMHO those should always be released, but it will skip one of them if it matches the current os/arch.

But let's start the other way around: do you even want to have a build with $GOHOSTOS / $GOHOSTARCH, or do you just always need the linux/amd64 version? That would make it more simple:
```
        eval "$(go env)"
        BIN_NAME=$(basename $arg)
        ARCH_BASENAME=${BIN_NAME}-$(git describe --always --tags)
        LINUX_NAME=${ARCH_BASENAME}-linux-amd64
        mkdir -p ${CMD_OUT_DIR}/${BIN_NAME}
        (
            cd $arg
            go vet ./...
            
            # always build linux/amd64
            GOOS=linux GOARCH=amd64 go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${LINUX_NAME}
            (cd ${CMD_OUT_DIR}/${BIN_NAME} && ln -sf ${LINUX_NAME} ${BIN_NAME})
            
            # build virtctl also for darwin and windows
            if [ "${BIN_NAME}" = "virtctl" ]; then
                GOOS=darwin GOARCH=amd64 go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-darwin-amd64
                GOOS=windows GOARCH=amd64 go build -o ${CMD_OUT_DIR}/${BIN_NAME}/${ARCH_BASENAME}-windows-amd64.exe
            fi
        )
```

Reply:
> But let's start the other way around: do you even want to have a build with $GOHOSTOS / $GOHOSTARCH, or do you just always need the linux/amd64 version? That would make it more simple:

I believe this simplification is what we need. cc @rmohr

Reply:
For the client side it does make sense to also support other architectures.